src/librustdoc/lib.rs

```diff
@@ -259,6 +259,7 @@ fn opts() -> Vec<RustcOptGroup> {
             o.optmulti("L", "library-path", "directory to add to crate search path", "DIR")
         }),
         stable("cfg", |o| o.optmulti("", "cfg", "pass a --cfg to rustc", "")),
+        unstable("check-cfg", |o| o.optmulti("", "check-cfg", "pass a --check-cfg to rustc", "")),
```

Comment:
Why doing this?
It's not rustdoc's job to report lints unrelated to documentation.

Reply:
In the previous PR, `@jyn514` wrote https://github.com/rust-lang/rust/pull/89346#issuecomment-930129761 :

> It would be useful, I think, it catches issues like cfg(doctst) or something (and in general I would like expansion to match rustc as closely as possible).

Reply:
Could you make a separate PR for this (after this one lands), put the motivation into its description, and then assign to someone from rustdoc team?